src/sagemaker_pytorch_serving_container/torchserve.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2019-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module contains functionality to configure and start the
+multi-model server."""
+from __future__ import absolute_import
+
+import os
+import signal
+import subprocess
+import sys
+
+import pkg_resources
+import psutil
+from retrying import retry
+
+import sagemaker_pytorch_serving_container
+from sagemaker_inference import default_handler_service, environment, logging, utils
+from sagemaker_inference.environment import code_dir
+
+logger = logging.get_logger()
+
+TS_CONFIG_FILE = os.path.join("/etc", "sagemaker-ts.properties")
+DEFAULT_HANDLER_SERVICE = default_handler_service.__name__
+DEFAULT_TS_CONFIG_FILE = pkg_resources.resource_filename(
+    sagemaker_pytorch_serving_container.__name__, "/etc/default-ts.properties"
+)
+MME_TS_CONFIG_FILE = pkg_resources.resource_filename(
+    sagemaker_pytorch_serving_container.__name__, "/etc/mme-ts.properties"
+)
+DEFAULT_TS_LOG_FILE = pkg_resources.resource_filename(
+    sagemaker_pytorch_serving_container.__name__, "/etc/log4j.properties"
+)
+DEFAULT_TS_MODEL_DIRECTORY = os.path.join(os.getcwd(), ".sagemaker/ts/models")
+DEFAULT_TS_MODEL_NAME = "model"
+DEFAULT_TS_MODEL_SERIALIZED_FILE = "model.pth"
+DEFAULT_HANDLER_SERVICE = "sagemaker_pytorch_serving_container.handler_service"
+
+ENABLE_MULTI_MODEL = os.getenv("SAGEMAKER_MULTI_MODEL", "false") == "true"
+MODEL_STORE = "/" if ENABLE_MULTI_MODEL else DEFAULT_TS_MODEL_DIRECTORY
+
+PYTHON_PATH_ENV = "PYTHONPATH"
+REQUIREMENTS_PATH = os.path.join(code_dir, "requirements.txt")
+TS_NAMESPACE = "org.pytorch.serve.ModelServer"
+
+
+def start_torchserve(handler_service=DEFAULT_HANDLER_SERVICE):
+    """Configure and start the model server.
+
+    Args:
+        handler_service (str): python path pointing to a module that defines
+            a class with the following:
+
+                - A ``handle`` method, which is invoked for all incoming inference
+                    requests to the model server.
+                - A ``initialize`` method, which is invoked at model server start up
+                    for loading the model.
+
+            Defaults to ``sagemaker_pytorch_serving_container.default_handler_service``.
+
+    """
+
+    if ENABLE_MULTI_MODEL:
+        if not os.getenv("SAGEMAKER_HANDLER"):
```

Comment:
if you're not expecting `"SAGEMAKER_HANDLER"` to ever be something like `""`, then it's probably more Pythonic to do:

```suggestion
        if "SAGEMAKER_HANDLER" not in os.environ:
```